Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -132,11 +135,19 @@ public class PythonCompiler : SinglePassCompiler {
       var btw = wr.NewBlockPy($"class {DtT}:");
 
       foreach (var ctor in dt.Ctors) {
+        var ctorName = MangleName(ctor.CompileName);
+        
         // Class-level fields don't work in all python version due to metaclasses.
-        var namedtuple = $"NamedTuple(\"{ctor.CompileName}\", [])";
+        var argList = ctor.Destructors.Select(d => $"(\'{ctorName}\', {TypeName(d.Type, wr, d.tok)})").Comma();
+        var namedtuple = $"NamedTuple(\"{ctorName}\", [{argList}])";
         var header = $"class {DtCtorDeclarationName(ctor, false)}({DtT}, {namedtuple}):";
         var constructor = wr.NewBlockPy(header, close: BlockStyle.Newline);
         DatatypeFieldsAndConstructor(ctor, constructor);
+        
+        // def is_Ctor0(self):
+        //   return isinstance(self, Dt_Ctor0) }
+        var query = btw.NewBlockPy($"def is_{ctorName}(self):");
+        query.WriteLine($"return isinstance(self, {ctorName})");
```

Comment:
I inlined this instead of the is_* call. Not sure what is better.

Reply:
https://github.com/dafny-lang/dafny/blob/65c1a012c09fdae8940753bbd4f0e7f7ac2e71df/Source/Dafny/Compilers/Compiler-python.cs#L896-L898

Reply:
Me neither. I suppose inlining it is slightly more efficient so I'll take your version instead.